cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1226 @@
+	incomplete         = "incomplete"
+)
+
+// Config is a format for storing update data
+type Config struct {
+	CardID      string         `json:"CardID"`
+	ChecklistID string         `json:"ChecklistID"`
+	Key         string         `json:"Key"`
+	Token       string         `json:"Token"`
+	Exchanges   []ExchangeInfo `json:"Exchanges"`
+}
+
+var (
+	verbose                                                                        bool
+	apiKey, apiToken, trelloBoardID, trelloListID, trelloChecklistID, trelloCardID string
+	configData, testConfigData                                                     Config
+)
+
+func main() {
+	flag.StringVar(&apiKey, "key", "", "sets API key for Trello board interaction")
+	flag.StringVar(&apiToken, "token", "", "sets API token for Trello board interaction")
+	flag.StringVar(&trelloChecklistID, "checklistid", "", "sets checklist ID for Trello board interaction")
+	flag.StringVar(&trelloCardID, "cardid", "", "sets card ID for Trello board interaction")
+	flag.StringVar(&trelloListID, "listid", "", "sets list ID for Trello board interaction")
+	flag.StringVar(&trelloBoardID, "boardid", "", "sets board ID for Trello board interaction")
+	flag.BoolVar(&verbose, "verbose", false, "increases logging verbosity for API Update Checker")
+	flag.Parse()
+	var err error
+	testConfigData, err = ReadFileData(testJSONFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	configData, err = ReadFileData(jsonFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if CanUpdateTrello() {
+		UpdateFile(&configData, backupFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		configData.Key = apiKey
+		configData.Token = apiToken
+		configData.CardID = trelloCardID
+		configData.ChecklistID = trelloChecklistID
+		err = CheckUpdates(jsonFile, &configData)
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		log.Printf("This is a test update since API keys are not set.\n")
+		err := CheckUpdates(testJSONFile, &testConfigData)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// CanUpdateTrello checks if all the data necessary for updating trello is available
+func CanUpdateTrello() bool {
+	if AreAPIKeysSet() && IsTrelloBoardDataSet() {
+		return true
+	}
+	return false
+}
+
+// AreAPIKeysSet checks if api keys and tokens are set
+func AreAPIKeysSet() bool {
+	if (apiKey != "" && apiToken != "") || (configData.Key != "" && configData.Token != "") {
+		return true
+	}
+	return false
+}
+
+// IsTrelloBoardDataSet checks if data required to update trello board is set
+func IsTrelloBoardDataSet() bool {
+	if (trelloBoardID != "" && trelloListID != "" && trelloChecklistID != "" && trelloCardID != "") ||
+		(configData.CardID != "" && configData.ChecklistID != "") {
+		return true
+	}
+	return false
+}
+
+// getSha gets the sha of the latest commit
+func getSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	path := fmt.Sprintf(githubPath, repoPath)
+	if verbose {
+		fmt.Printf("Getting SHA of this path: %v\n", path)
```

Comment:
Please use `log.Printf` as you get inconsistent ouput:
```
Getting SHA of this path: https://api.github.com/repos/bithumb-pro/bithumb.pro-o                                                                                                                fficial-api-docs/commits/master
2020/02/09 22:29:21 Updating test file, main file & trello will not be automatically updated since API key & token are not set
```

Date missing